rosidl_adapter/rosidl_adapter/cli.py

```diff
@@ -48,3 +51,57 @@ def convert_files_to_idl(extension, conversion_function, argv=sys.argv[1:]):
             package_dir, pkg.name,
             interface_file.absolute().relative_to(package_dir),
             interface_file.parent)
+
+
+class TranslateToIDL(TranslateCommandExtension):
+
+    output_format = 'idl'
+
+    def translate(
+        self,
+        package_name,
+        interface_files,
+        include_paths,
+        output_path
+    ):
+        translated_interface_files = []
+        for interface_file in interface_files:
+            prefix, interface_file = interface_path_as_tuple(interface_file)
+            output_dir = output_path / interface_file.parent
+            translated_interface_file = self.conversion_function(
+                prefix, package_name, interface_file, output_dir)
+            translated_interface_file = \
+                translated_interface_file.relative_to(output_path)
+            translated_interface_files.append(
+                f'{output_path}:{translated_interface_file}'
+            )
+        return translated_interface_files
+
+
+class TranslateMsgToIDL(TranslateToIDL):
+
+    input_format = 'msg'
+
+    @property
+    def conversion_function(self):
+        from rosidl_adapter.msg import convert_msg_to_idl
```

Comment:
Why is the import in the function instead of on the module?

Reply:
This source file didn't import them directly, so I just went with it. But I don't mind strongly. Moved to the top in https://github.com/ros2/rosidl/pull/576/commits/74e0044bdabae5c6f25ce76c6f591423a26ef4ee.